src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/LoggerMessageDefineAnalyzer.cs

```diff
@@ -0,0 +1,314 @@
+                                                                         RuleLevel.Disabled,
+                                                                         description: s_localizableDescriptionCA1848,
+                                                                         isPortedFxCopRule: false,
+                                                                         isDataflowRule: false,
+                                                                         isReportedAtCompilationEnd: false);
+
+        internal static DiagnosticDescriptor CA2253Rule = DiagnosticDescriptorHelper.Create(CA2253RuleId,
+                                                                         s_localizableTitleCA2253,
+                                                                         s_localizableMessageCA2253,
+                                                                         DiagnosticCategory.Usage,
+                                                                         RuleLevel.IdeSuggestion,
+                                                                         description: s_localizableDescriptionCA2253,
+                                                                         isPortedFxCopRule: false,
+                                                                         isDataflowRule: false,
+                                                                         isReportedAtCompilationEnd: false);
+
+        internal static DiagnosticDescriptor CA2254Rule = DiagnosticDescriptorHelper.Create(CA2254RuleId,
+                                                                         s_localizableTitleCA2254,
+                                                                         s_localizableMessageCA2254,
+                                                                         DiagnosticCategory.Usage,
+                                                                         RuleLevel.IdeSuggestion,
+                                                                         description: s_localizableDescriptionCA2254,
+                                                                         isPortedFxCopRule: false,
+                                                                         isDataflowRule: false,
+                                                                         isReportedAtCompilationEnd: false);
+
+        internal static DiagnosticDescriptor CA2255Rule = DiagnosticDescriptorHelper.Create(CA2255RuleId,
+                                                                         s_localizableTitleCA2255,
+                                                                         s_localizableMessageCA2255,
+                                                                         DiagnosticCategory.Usage,
+                                                                         RuleLevel.BuildWarning,
```

Comment:
You should not use `RuleLevel.BuildWarning`. It is reserved for highly critical analyzers that indicate serious functional bug in code, something that is guaranteed to make the code behave badly at runtime. These are considered build breaking rules as /warnaserror would cause anyone updating to new .NET SDK to break their builds, something we never want for naming or usage or performance rules

Reply:
the API review approved the fifth diagnostic as warning since its usage would result in a runtime exception